### ISSUE TITLE
Prompt logprobs misaligned: left shift needed; BOS logprob must be NaN

### ISSUE THREAD

**Issue by babyplutokurt** [AI-Hypercomputer/maxtext#2103]
### Bug report

Observed **very large negative logprobs across all prompt positions**. 
Root cause: in maxengine.py prefill() line 478: 
`prompt_logp = inference_utils.log_prob_of_chosen_token(flat_logits, input_tokens)`
logprobs were computed as **log P(token[t] | tokens[0..t])** (same-position logits), rather than **log P(token[t] | tokens[0..t-1])**. This off-by-one alignment makes real tokens look extremely unlikely.

### Details
- Decoder logits at step *i* predict token *i+1*. To score token *t*, we must use **logits[t-1]**, i.e., shift by one.
- The first prompt position (BOS) has no preceding context, so its logprob must be **null/NaN** (OpenAI semantics).
- Padding/final positions should not be scored; the final prefill logit predicts the **first completion token**, not a prompt token.

### Expected behavior
- Return per-token prompt logprobs: `log P(token[t] | tokens[:t])` for `t = 1..true_length-1`.
- `logprobs[0] = NaN` (BOS).
- Positions `>= true_length` are ignored (NaN).

### Fix outline
- Left-shift targets: score tokens `1..S-1` using logits `0..S-2`.
- Mask `t=0` (BOS) and any `t >= true_length` to NaN.
- Keep using the final prefill logit (`t = true_length-1`) **only** to score the **first generated token**.

### Input Prompt: Hello, who are you?  -->` ['', 'Hello', ',', 'who', 'are', 'you', '?']`
Before: Prompt logprobs
<img width="708" height="603" alt="image" src="https://github.com/user-attachments/assets/0eb6aa67-fabe-4c2b-8068-2b0581cb4e84" />
